### PR TITLE
refactor: introduce decoupled platform layer and convert models to state objects

### PR DESCRIPTION
**Pull request by chr-hertel**
this PR focusses on shaping the model and platform concept.

* models are only state objects now and the service character got killed in favor of Platform extension points `ResponseFactory` and `ResponseConverter` => decoupling => way better testability
* opens up the platform requests to be quite generic => leeway to supporting large platforms like ollama, huggingface or  replicate in a generic manner
* bridge namespace heavily inspired by Symfony

Only thing missing right now is fixing embedding by bringing support for parallel calls on Platform back

_best to review in IDE_

### CODE REVIEW

File: src/Bridge/Azure/OpenAI/PlatformFactory.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        #[\SensitiveParameter] string $apiKey,
+    ): Platform {
+        $httpClient = new EventSourceHttpClient();
+        $embeddingsResponseFactory = new EmbeddingsResponseFactory($httpClient, $baseUrl, $deployment, $apiVersion, $apiKey);
+        $GPTResponseFactory = new GPTResponseFactory($httpClient, $baseUrl, $deployment, $apiVersion, $apiKey);
+
+        return new Platform(
+            [$GPTResponseFactory, $embeddingsResponseFactory],
+            [new GPT\ResponseConverter(), new Embeddings\ResponseConverter()],
+        );
```
Comment (chr-hertel) [inline src/Bridge/Azure/OpenAI/PlatformFactory.php:27]: one important benefit:
response factory for the Azure specific platform is different, but response factory can sit on the openai default one

**Comment by OskarStark** [comment]
Sorry for being not so active, I am a little bit busy right now.
I didn't find time to check this PR in detail

**Comment by chr-hertel** [comment]
@OskarStark no worries at all Oskar - i'm currently finishing up those 4 MRs and will try them in my demo app with the bundle to get them merged :+1:

**Comment by OskarStark** [comment]
Thanks, if my response format is working for tools, I could also switch entirely to our bundle/lib

**Pull request merged**